homeassistant/components/zwave_js/api.py

```diff
@@ -183,6 +243,56 @@ async def async_handle_failed_command_func(
     return async_handle_failed_command_func
 
 
+def get_provisioning_info(
+    msg: dict,
+) -> str | ProvisioningEntry | QRProvisioningInformation | None:
+    """Process provisioning parameters and return the appropriate value."""
+    if planned_provisioning_entry := msg.get(PLANNED_PROVISIIONING_ENTRY):
+        return ProvisioningEntry(
+            dsk=planned_provisioning_entry[DSK],
+            security_classes=[
+                SecurityClass(sec_cls)
+                for sec_cls in planned_provisioning_entry[SECURITY_CLASSES]
+            ],
+            additional_properties={
+                k: v
+                for k, v in planned_provisioning_entry.items()
+                if k not in (DSK, SECURITY_CLASSES)
+            },
+        )
+
+    if qr_provisioning_information := msg.get(QR_PROVISIONING_INFORMATION):
+        protocols = [
+            Protocols(proto)
+            for proto in qr_provisioning_information.get(SUPPORTED_PROTOCOLS, [])
+        ]
+        return QRProvisioningInformation(
+            version=QRCodeVersion(qr_provisioning_information[VERSION]),
```

Comment:
Before starting inclusion or SmartStart provisioning we need to check this QR code version and see if it's an S2 or SmartStart code.

https://zwave-js.github.io/node-zwave-js/#/api/controller?id=begininclusion

https://zwave-js.github.io/node-zwave-js/#/api/controller?id=provisionsmartstartnode

The frontend should probably check this by parsing the QR code with that api helper, but should we fail in the the inclusion and provisioning api functions if there's a mismatch?